### PR TITLE
feat(ui): Adds related issues to Incidents sidebar [SEN-695]

### PR DESCRIPTION
**Pull request by billyvg**
Adds related issues to sidebar

image

Fixes SEN-695

### CODE REVIEW

File: src/sentry/static/sentry/app/views/organizationIncidents/details/relatedIssues/issuesFetcher.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    if (prevProps.issueIds !== this.props.issueIds) {
+      this.fetchData();
+    }
+  }
+
+  fetchData = async () => {
+    const {api, issueIds} = this.props;
+    this.setState({loading: true});
+
+    if (!issueIds) {
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/organizationIncidents/details/relatedIssues/issuesFetcher.jsx]: If there are no issues, how does this component leave its loading state?

File: tests/js/spec/views/organizationIncidents/details/index.spec.jsx [hunk 2]
```diff
@@ -160,4 +158,50 @@ describe('IncidentDetails', function() {
     wrapper.find('SubscribeButton').simulate('click');
     expect(subscribe).toHaveBeenCalled();
   });
+
+  it('loads related incidents', async function() {
```
Comment (markstory) [inline tests/js/spec/views/organizationIncidents/details/index.spec.jsx]:
```suggestion
  it('loads related issues', async function() {
```

File: src/sentry/static/sentry/app/views/organizationIncidents/details/relatedIssues/issuesFetcher.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationIncidents/details/relatedIssues/issuesFetcher.jsx]: Oops, thanks

File: src/sentry/static/sentry/app/views/organizationIncidents/details/relatedIssues/index.jsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+              return (
+                <React.Fragment>
+                  <SideHeader loading={loading}>
+                    {t('Related Issues')} ({loading || !issues ? '-' : issues.length})
+                  </SideHeader>
+                  {loading ? (
+                    <Placeholder />
+                  ) : (
+                    issues &&
+                    issues.length > 0 && (
```
[the remark this answers is not in the document]
Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationIncidents/details/relatedIssues/index.jsx]: I've changed to make sure entire block does not show (including header) - will follow up to see if this is what we want.

File: src/sentry/static/sentry/app/views/organizationIncidents/details/relatedIssues/index.jsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    };
+
+    render() {
+      const {className, api, params, incident} = this.props;
+
+      return (
+        <div className={className}>
+          <IssuesFetcher api={api} issueIds={incident && incident.groups}>
+            {({issues, loading, error}) => {
+              // If loading is finished, and there are no issues, do not display anythig
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/organizationIncidents/details/relatedIssues/index.jsx]:
```suggestion
              // If loading is finished, and there are no issues, do not display anything
```

File: src/sentry/static/sentry/app/views/organizationIncidents/details/relatedIssues/issuesFetcher.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationIncidents/details/relatedIssues/issuesFetcher.jsx]: Fixed by moving where we set loading state: https://github.com/getsentry/sentry/pull/13617/files#diff-488f70254ef5984b677ed6927ad48ca9R33

**Review by markstory — APPROVED** [review]

**Pull request merged**